src/ensure-specs-running/main.go

```diff
@@ -0,0 +1,212 @@
+            strings.Join(actualDeployments.heapster.containers, ", "),
+        ))
+    }
+    if expectedDeployments.influxdb.replicas != actualDeployments.influxdb.replicas {
+        problems = append(problems, fmt.Sprintf(
+            "expected %d %s replicas, found %d",
+            expectedDeployments.influxdb.replicas,
+            influxdb,
+            actualDeployments.influxdb.replicas,
+        ))
+    }
+    if !match(expectedDeployments.influxdb.containers, actualDeployments.influxdb.containers) {
+        problems = append(problems, fmt.Sprintf(
+            "expected %s to have [%s] containers running, but found [%s] containers running",
+            influxdb,
+            strings.Join(expectedDeployments.influxdb.containers, ", "),
+            strings.Join(actualDeployments.influxdb.containers, ", "),
+        ))
+    }
+    if expectedDeployments.dashboard.replicas != actualDeployments.dashboard.replicas {
+        problems = append(problems, fmt.Sprintf(
+            "expected %d %s replicas, found %d",
+            expectedDeployments.dashboard.replicas,
+            dashboard,
+            actualDeployments.dashboard.replicas,
+        ))
+    }
+    if !match(expectedDeployments.dashboard.containers, actualDeployments.dashboard.containers) {
+        problems = append(problems, fmt.Sprintf(
+            "expected %s to have [%s] containers running, but found [%s] containers running",
+            dashboard,
+            strings.Join(expectedDeployments.dashboard.containers, ", "),
+            strings.Join(actualDeployments.dashboard.containers, ", "),
+        ))
+    }
+    if expectedDeployments.dns.replicas != actualDeployments.dns.replicas {
+        problems = append(problems, fmt.Sprintf(
+            "expected %d %s replicas, found %d",
+            expectedDeployments.dns.replicas,
+            dns,
+            actualDeployments.dns.replicas,
+        ))
+    }
+    if !match(expectedDeployments.dns.containers, actualDeployments.dns.containers) {
+        problems = append(problems, fmt.Sprintf(
+            "expected %s to have [%s] containers running, but found [%s] containers running",
+            dns,
+            strings.Join(expectedDeployments.dns.containers, ", "),
+            strings.Join(actualDeployments.dns.containers, ", "),
+        ))
+    }
+
+    if len(problems) != 0 {
+        dumpPods(pods.Items)
+        log.Fatal("problems found:\n- " + strings.Join(problems, "\n- "))
+    }
+}
+
+func extractDeployment(pod corev1.Pod) string {
+    if _, hasPodTemplateHash := pod.ObjectMeta.Labels["pod-template-hash"]; !hasPodTemplateHash {
+        return unrecognized
+    }
+
+    numLabels := len(pod.ObjectMeta.Labels)
+    task := pod.ObjectMeta.Labels["task"]
+    k8sApp := pod.ObjectMeta.Labels["k8s-app"]
+    version := pod.ObjectMeta.Labels["version"]
+
+    if numLabels == 4 && task == "monitoring" && k8sApp == "heapster" && version == "v6" {
```

Comment:
Also it would be nice to use the constant strings you created above (on line 45) for "heapster", "influxdb", etc